internal/image/build.go

```diff
@@ -0,0 +1,175 @@
+	Invocation  invocation             `json:"invocation"`
+	BuildType   string                 `json:"buildType"`
+	Metadata    map[string]interface{} `json:"metadata"`
+	Builder     map[string]interface{} `json:"builder"`
+	BuildConfig map[string]interface{} `json:"buildConfig"`
+	Materials   []materials            `json:"materials"`
+}
+
+type attestation struct {
+	Predicate     predicate                `json:"predicate"`
+	PredicateType string                   `json:"predicateType"`
+	Subject       []map[string]interface{} `json:"subject"`
+	Type          string                   `json:"_type"`
+}
+
+type commitSignOff struct {
+	source    string
+	commitSha string
+}
+
+// there can be multiple sign off sources (git commit, tag and jira issues)
+type signOffSource interface {
+	GetSignOff() (*signOffSignature, error)
+}
+
+type commit struct {
+	Sha     string `json:"sha"`
+	Author  string `json:"author"`
+	Date    string `json:"date"`
+	Message string `json:"message"`
+}
+
+type signOffSignature struct {
+	Body       interface{} `json:"body"`
+	Signatures []string    `json:"signatures"`
+}
+
+// From an attestation, find the signOff source (commit, tag, jira)
+func (a *attestation) NewSignOffSource() (signOffSource, error) {
+	// the signoff source can be determined by looking into the attestation.
+	// the attestation can have an env var or something that this can key off of
+
+	commitSha := a.getBuildCommitSha()
+	repo := a.getBuildSCM()
+	if commitSha != "" && repo != "" {
+		return &commitSignOff{
+			source:    a.getBuildSCM(),
+			commitSha: commitSha,
+		}, nil
+	}
+	return nil, nil
+}
+
+// get the last commit used for the component build
+func (a *attestation) getBuildCommitSha() string {
+	if len(a.Predicate.Materials) == 1 {
+		return a.Predicate.Materials[0].Digest["sha"]
+	}
+	return ""
+}
+
+// the git url used for the component build
+func (a *attestation) getBuildSCM() string {
+	if len(a.Predicate.Materials) == 1 {
+		return a.Predicate.Materials[0].Uri
+	}
+	return ""
+}
+
+// returns the signOff signature and body of the source
+func (c *commitSignOff) GetSignOff() (*signOffSignature, error) {
+	commit, err := getCommitSource(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return &signOffSignature{
+		Body:       commit,
+		Signatures: captureCommitSignOff(commit.Message),
+	}, nil
+}
+
+// get the build commit source for use in GetSignOff
+func getCommitSource(data *commitSignOff) (*commit, error) {
+	repo, err := gitClone(memory.NewStorage(), nil, &git.CloneOptions{
+		URL: data.source,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	gitCommit, err := repo.CommitObject(plumbing.NewHash(data.commitSha))
+	if err != nil {
+		return nil, err
+	}
+
+	return &commit{
+		Sha:     gitCommit.Hash.String(),
+		Author:  fmt.Sprintf("%s <%s>", gitCommit.Author.Name, gitCommit.Author.Email),
+		Date:    gitCommit.Author.When.String(),
+		Message: gitCommit.Message,
+	}, nil
+
+}
+
+// parse a commit and capture signatures
```

Comment:
Let's update this comment to say "sign offs" instead of "signatures". Same for the comments within this function.